### PR TITLE
Update pytorch/functorch versions

### PR DESCRIPTION
**Pull request by ezyang**
functorch is now part of pytorch/pytorch repo, so we now prefer
that to ensure we have working versions in sync.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]

**Review by voznesenskym — APPROVED** [review]
LGTM

File: torchdynamo/optimizations/python_key.py [hunk 1]
```diff
@@ -160,6 +157,7 @@ def unpack(x):
 
     with pythonkey_decompose({**aot_autograd_decompositions, **decompositions}):
         tracer: torch.fx.Tracer = PythonKeyTracer()
+        proxy_mode = ProxyTorchDispatchMode(tracer)
```
Comment (ezyang) [inline torchdynamo/optimizations/python_key.py:160]: We'll merge this as is. But can someone explain to me why we have this duplicate implementation of `make_fx` here? Can we dump this code and use most of `make_fx` to do this?

Comment (voznesenskym) [reply torchdynamo/optimizations/python_key.py:160]: I don't have history, but @Chillee and I were talking about just merging most of the stuff in here with the python key tracing in fx.

Comment (ezyang) [reply torchdynamo/optimizations/python_key.py:160]: yes plz

**Comment by voznesenskym** [comment]
More doozy than expected, a few test failures. Looking closer here...

File: tests/test_optimizers.py [hunk 2]
```diff
@@ -12,14 +12,14 @@
...
-def make_test(optim_cls, exp_frame_cnt=1, **kwargs):
+def make_test(optim_cls, exp_frame_cnt=1, nopython=True, **kwargs):
     opt = optim_cls(model.parameters(), **kwargs)
 
     def test_fn(self):
         nonlocal opt
         counter = torchdynamo.testing.CompileCounter()
 
-        with torchdynamo.optimize(counter):
+        with torchdynamo.optimize(counter, nopython=nopython):
```
Comment (mlazos) [inline tests/test_optimizers.py:22]: If you remove nopython=True, Adam will continue to work. I just tried this locally.

**Comment by voznesenskym** [comment]
https://github.com/pytorch/pytorch/issues/82763

File: torchinductor/lowering.py [hunk 3]
```diff
@@ -1023,13 +1023,14 @@ def _local_scalar_dense(data):
 
 
 def _full(fill_value, device, dtype, size):
-    if isinstance(fill_value, ir.Constant):
-        fill_value = fill_value.value
-    assert isinstance(fill_value, (int, float)), fill_value
+    value = fill_value
+    if not isinstance(fill_value, (int, float)):
+        value = value.value
+    assert isinstance(value, (int, float)), f"Expected int or float, got {value}"
```
Comment (msaroufim) [inline torchinductor/lowering.py:1029]:
```suggestion
    assert isinstance(value, (int, float)), f"Expected int or float, got type {type(value)}"
```

File: torchdynamo/variables/torch.py [hunk 4]
```diff
@@ -128,6 +129,8 @@ def call_function(
                 return ConstantVariable(True, **options)
             elif self.value is torch.is_floating_point:
                 return ConstantVariable(args[0].dtype.is_floating_point, **options)
+            elif self.value is torch.is_complex:
```
Comment (msaroufim) [inline torchdynamo/variables/torch.py:132]: Is this all we support, what about things like quantized `qint`?

**Review by msaroufim — APPROVED** [review]

File: torchdynamo/variables/torch.py (hunk 4 above)
Comment (voznesenskym) [reply torchdynamo/variables/torch.py:132]: This pr is for getting past bugs and errors in bumping versions as per the title. If you want to request a torchdynamo feature please open a PR!

File: torchdynamo/optimizations/python_key.py (hunk 1 above)
Comment (jansel) [reply torchdynamo/optimizations/python_key.py:160]: More context is here https://github.com/pytorch/torchdynamo/issues/271

This version was originally created to capture factory functions, which at the time AOT Autograd didn't support.

**Review by jansel — APPROVED** [review]

**Comment by voznesenskym** [comment]
Waiting on `torch==1.13.0.dev20220805`

**Comment by voznesenskym** [comment]
Initial errors w/ nan_to_num solved, but new ones hit around arg promotion. https://github.com/pytorch/pytorch/issues/82920

**Comment by Chillee** [comment]
Would be nice to get this in, could we just skip the failing tests for now?

**Pull request merged**